server/questions.js

```diff
@@ -21,7 +21,17 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 var db = require('./db.js');
 var logger = require('./log.js').logger;
 var common = require('./common.js');
+var validator = require('./validation.js');
 
+var prepareQuestionData = function(data){
```

Comment:
this is a start of a great idea, I love where this is going. Lets have prepareQuestionData(data, callback) In this function we will do the validation and if t he validation passes we can return a pure object of what the question needs to be created/edited. For this we will need different prepare function for each question type. So like, prepareMultipleChoiceQuestionData(data, callback) and so on.

Reply:
That is a good point, what we can do here is, where I have created functions for each question type to validate (i.e. multipleChoiceValidator,trueAndFalseValidator) We can have it prepare the data and validate it as well. This saves extra code from being created.